types/mongoose/index.d.ts

```diff
@@ -2000,7 +2000,7 @@ declare module "mongoose" {
      * getters/setters or other Mongoose magic applied.
      * @param {Boolean|Object} bool defaults to true
      */
-    lean(bool?: boolean | object): Query<any> & QueryHelpers;
+    lean<T>(bool?: boolean | object): Query<T> & QueryHelpers;
```

Comment:
Shouldn't it be `lean(bool?: boolean | object): Query<T> & QueryHelpers;` ? (without the 1st <T>)
This seems to make all results `unknown` right now.

Reply:
Yes of course. It is semantically more correct than type `any`.

Reply:
Indeed but what I meant is we do have the type, lean could be correctly (or as correctly as can be) typed instead of returning unknown.